controllers/rollingupgrade_controller.go

```diff
@@ -872,25 +892,38 @@ func (r *RollingUpgradeReconciler) UpdateInstances(ctx *context.Context,
 	ch := make(chan error)
 
 	for _, instance := range instances {
+		// log it before we start updating the instance
+		r.createK8sV1Event(ruObj, EventReasonRUInstanceStarted, EventLevelNormal, map[string]string{
+			"status":   "in-progress",
+			"asgName":  ruObj.Spec.AsgName,
+			"strategy": string(ruObj.Spec.Strategy.Type),
+			"msg":      fmt.Sprintf("Started Updating Instance %s, in AZ: %s", *instance.InstanceId, *instance.AvailabilityZone),
+		})
 		go r.UpdateInstance(ctx, ruObj, instance, launchDefinition, KubeCtlCall, ch)
 	}
 
 	// wait for upgrades to complete
 	nodesProcessed := 0
 	var instanceUpdateErrors []error
-Loop:
+
 	for err := range ch {
```

Comment:
wont this get stuck until channel is closed?

Reply:
`if nodesProcessed == totalNodes { break }` should break the loop if we have `nodesProcessed == totalNodes`